R/read_data.R

```diff
@@ -34,6 +34,19 @@ read_data.shapedir <- function(filename, ...){
 
 #' @export
 #' @keywords internal
+read_data.gzfile <- function(filename, ...){
+  file.con <- gzfile(filename)
+  if (grepl('tsv', filename)){
+    read_data.tsvfile(file.con, ...)
+  } else if (grepl('csv', filename)){
+    read_data.csvfile(file.con, ...)
+  } else {
+    stop('type ', filename, ' not recognized by read_data', call. = FALSE)
+  }
+  close(file.con)
+}
+#' @export
+#' @keywords internal
 read_data.csvfile <- function(filename, ...){
```

Comment:
agreed - not really trying to limit deps because this is an internal package (and maybe deps matter less because of that)

but...these are really small files because they only have a row for each attribute of the file (shapefile or tabular file, etc). Like here: https://github.com/USGS-CIDA/stream_metab_usa/blob/master/4_data_release/in/catchment_attr.csv